// dicebear/lib/utils/getStyleCommandSchema.js
import { schema as coreSchema } from '@dicebear/core';
import mergeAllOf from 'json-schema-merge-allof';
export function getStyleCommandSchema(style) {
    var _a;
    return mergeAllOf({
        allOf: [
            {
                properties: {
                    count: {
                        description: 'Defines how many avatars to create.',
                        type: 'number',
                        default: 1,
                    },
                    format: {
                        type: 'string',
                        enum: ['svg', 'png', 'jpg', 'jpeg', 'json'],
                        default: 'svg',
                    },
                    exif: {
                        description: 'Include Exif Metadata',
                        type: 'boolean',
                        default: false,
                    },
                    json: {
                        description: 'Save JSON file in addition to image file',
                        type: 'boolean',
                        default: false,
                    }
                },
            },
            coreSchema,
            (_a = style.schema) !== null && _a !== void 0 ? _a : {},
        ],
        additionalItems: true,
    }, { ignoreAdditionalProperties: true });
}
